utils/models_setup.py

```diff
@@ -8,9 +9,47 @@
 from models.MFCStack import MFCStack
 
 
-def generate_model(model_type: str, input_dim: int, dataset: [Data], config: dict):
+def get_gpus_list():
+
+    available_gpus = []
+    available_gpus = [i for i in range(torch.cuda.device_count())]
+
+    return available_gpus
+
+
+def get_gpu(number):
+    gpus_list = get_gpus_list()
+    if number not in gpus_list:
+        raise ValueError(
+            "The GPU ID:" + str(number) + " is not inside the list of GPUs available"
+        )
+    else:
+        device = torch.device(
+            "cuda:" + str(number)
+        )  # you can continue going on here, like cuda:1 cuda:2....etc.
+
+    return device
+
+
+def generate_model(
+    model_type: str,
+    input_dim: int,
+    dataset: [Data],
+    config: dict,
+    distributed_data_parallelism: bool = False,
+):
+
+    if distributed_data_parallelism:
+        world_size = os.environ["OMPI_COMM_WORLD_SIZE"]
+        world_rank = os.environ["OMPI_COMM_WORLD_RANK"]
+
     torch.manual_seed(0)
-    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+    available_gpus = get_gpus_list()
+    if len(available_gpus) > 0:
+        device = get_gpu(int(world_rank) % len(available_gpus))
```

Comment:
I think that calculation should be done inside `get_gpu`

Reply:
get_gpu returns the torch.device object associated with a specific number ID, assuming that the number ID is in the liste of the available CUDA devices, otherwise it returns an error. 
In view of a potential extension of the code where a single neural network does not fit on a single GPU, what you propose may be limiting.